hw/ip/otbn/dv/otbnsim/sim/csr.py

```diff
@@ -0,0 +1,64 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List
+
+from riscvmodel.types import Trace  # type: ignore
+
+from .flags import FlagGroups
+from .wsr import WSRFile
+
+
+class CSRFile:
+    '''A model of the CSR file'''
+    def __init__(self) -> None:
+        self.flags = FlagGroups()
+
+    def read_unsigned(self, wsrs: WSRFile, idx: int) -> int:
```

Comment:
I'm actually working on a change to the register file code to match this (doing the change the other way around). The point is that we might have read-sensitive registers (`x1`, for example), so we want to capture the act of reading them, rather than just talk about their value.

Re the memory, since it doesn't know its "width", we sort of need "read 32 bits" and a "read 256 bits" as separate operations, hence `read_u32` and `read_u256`. I'm not sure we can do better than that. Or have I misunderstood the last sentence?

Reply:
On the last sentence: with "memory" I meant "memory elements" in general, in this case, the WSRs. Your answer covers it all.